dotlottie-sm/src/state_machine.rs

```diff
@@ -0,0 +1,142 @@
+use std::sync::{Arc, RwLock};
+
+use dotlottie_player_core::DotLottiePlayer;
+
+use crate::{event::Event, state::State, state::StateTrait, transition::TransitionTrait};
+
+pub struct StateMachine {
+    pub states: Vec<Box<State>>,
+    pub current_state: Arc<RwLock<State>>,
+}
+
+impl StateMachine {
+    pub fn start(&mut self, player: &mut DotLottiePlayer) {
+        self.execute_current_state(player)
+    }
+
+    pub fn pause(&mut self) {}
+
+    pub fn end(&mut self) {}
+
+    pub fn set_initial_state(&mut self, state: Arc<RwLock<State>>) {
+        self.current_state = state;
+    }
+
+    // pub fn set_player(&mut self, player: &'a mut DotLottiePlayer) {
+    //     self.dotlottie_player = Some(player)
+    // }
+
+    // pub fn add_state(&mut self, state: Box<dyn State>) {
+    //     self.states.push(state);
+    // }
+
+    // pub fn execute_all_states(&mut self, player: &mut DotLottiePlayer) {
+    //     self.states.iter_mut().for_each(|state| {
+    //         state.execute(player);
+    //     });
+    // }
+
+    pub fn execute_current_state(&mut self, player: &mut DotLottiePlayer) {
+        let current_state = self.current_state.clone();
+        let state_value_result = current_state.write();
+
+        if state_value_result.is_ok() {
+            state_value_result.unwrap().execute(player);
+        }
+    }
+
+    pub fn post_event(&mut self, event: &Event) {
```

Comment:
[suggestion]

not important, maybe `emit_event`, or `trigger_event`, `fire_event` ?

Reply:
this is for the users to post and event to the state machine, I feel like those names feel like they'd be going from the state machine outwards. QT uses `postEvent`